src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/ReferenceNumberGeneratorService.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.companieshouse.api.strikeoffobjections.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.strikeoffobjections.service.IReferenceNumberGeneratorService;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+@Service
+public class ReferenceNumberGeneratorService implements IReferenceNumberGeneratorService {
+
+    private static final String REFERENCE_NUMBER_PREFIX = "OBJ-";
+
+    private final Supplier<LocalDateTime> dateTimeSupplier;
+
+    @Autowired
+    public ReferenceNumberGeneratorService(Supplier<LocalDateTime> dateTimeSupplier) {
+        this.dateTimeSupplier = dateTimeSupplier;
+    }
+
+    @Override
+    public String generateReferenceNumber() {
```

Comment:
I think some Javadoc with an example of the kind of reference number this method will generate would help interpret the code.